velox/functions/prestosql/WordStem.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <libstemmer.h>
+
+#include "velox/functions/Udf.h"
+#include "velox/functions/lib/string/StringImpl.h"
+
+namespace facebook::velox::functions {
+
+namespace {
+/// Wrap the sbstemmer library and use its sb_stemmer_stem
+/// to get word stem
+class Stemmer {
+ private:
+  sb_stemmer* sbStemmer_;
+
+ public:
+  Stemmer(sb_stemmer* stemmer) : sbStemmer_(stemmer) {}
+  ~Stemmer() {
+    sb_stemmer_delete(sbStemmer_);
+  }
+
+  /// Get the word stem or NULL if out of memory
+  const char* stem(const std::string& input) {
```

Comment:
Just curious, does this function return zero-terminated string?

Reply:
yes!

Reply:
Got it. Since this is a bit unusual in modern C++, it would be nice to clarify that in the comments.